storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2920,7 +2920,7 @@ private SupervisorSummary makeSupervisorSummary(String supervisorId, SupervisorI
                                                       (int) info.get_uptime_secs(), numPorts, numUsedPorts, supervisorId);
         ret.set_total_resources(info.get_resources_map());
         SupervisorResources resources = nodeIdToResources.get().get(supervisorId);
-        if (resources != null) {
+        if (resources != null && underlyingScheduler instanceof ResourceAwareScheduler) {
```

Comment:
What's the issue we want to solve here? 

If it is just about skipping the code related to resources when the scheduler is not resourceAware, then we have other places like `getTopologyPageInfo`  we need to take care of

Reply:
I have updated instances where code should avoid following resource calculation if not using `ResourceAwareScheduler`.